java-client/barrage/src/main/java/io/deephaven/client/impl/BarrageSubscriptionImpl.java

```diff
@@ -207,83 +180,37 @@ public BarrageTable partialTable(RowSet viewport, BitSet columns, boolean revers
     }
 
     @Override
-    public synchronized BarrageTable partialTable(RowSet viewport, BitSet columns, boolean reverseViewport,
+    public BarrageTable partialTable(RowSet viewport, BitSet columns, boolean reverseViewport,
             boolean blockUntilComplete) throws InterruptedException {
-        if (!connected) {
-            throw new UncheckedDeephavenException(
-                    this + " is no longer an active subscription and cannot be retained further");
-        }
-        if (subscribed) {
-            throw new UncheckedDeephavenException(
-                    "BarrageSubscription objects cannot be reused.");
+        synchronized (this) {
+            if (!connected) {
+                throw new UncheckedDeephavenException(
+                        this + " is no longer an active subscription and cannot be retained further");
+            }
+            if (subscribed) {
+                throw new UncheckedDeephavenException(
+                        "BarrageSubscription objects cannot be reused.");
+            }
+            subscribed = true;
         }
-        subscribed = true;
+
+        checkForCompletion.expectedViewport = viewport == null ? null : viewport.copy();
```

Comment:
Maybe we want to just construct `checkForCompletion` with these here, instead of in the constructor?

Reply:
The reason that I'm constructing in the constructor is because I would like to pass the "container" of callbacks to the BarrageTable.make() call. However, we don't yet know what the user will subscribe to; so we set the expected viewport once we know the actual subscription request.

Alternatively, I could add a method to BarrageTable to set the vpChangedCallback. If you have a strong preference to change to this, let me know. Otherwise, I will leave this comment alone.